collections/unzip.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Builds two separate arrays from the given array of 2-tuples, with the first returned array holding all first
+ * tuple elements and the second one holding all the second elements
+ *
+ * Example:
+ *
+ * ```typescript
+ * const parents = [
+ *     [ 'Maria', 'Jeff' ],
+ *     [ 'Anna', 'Kim' ],
+ *     [ 'John', 'Leroy' ],
+ * ]
+ * const [ moms, dads ] = unzip(parents)
+ *
+ * console.assert(moms === [ 'Maria', 'Anna', 'John' ])
+ * console.assert(moms === [ 'Jeff', 'Kim', 'Leroy' ])
+ * ```
+ */
+export function unzip<T, U>(pairs: Array<[T, U]>): [Array<T>, Array<U>] {
+  const { length } = pairs;
+  const ret: [Array<T>, Array<U>] = [
+    new Array(length),
+    new Array(length),
+  ];
+
+  pairs.forEach(([first, second], index) => {
```

Comment:
Feels a bit weird to have `.forEach` here when we're using `for...of` everywhere else, is there a reason to use it specifically here?

Reply:
Because I need the index and the elements, so forEach seemed most pragmatic. Could use entries and for of, but that creates an unnecessary iteration within entries.